src/current.rs

```diff
@@ -34,23 +40,74 @@ pub fn clock_for(task_id: TaskId) -> VectorClock {
     ExecutionState::with(|state| state.get_clock(task_id).clone())
 }
 
+/// Apply the given function to the Labels for the specified task
+pub fn with_labels_for_task<F, T>(task_id: TaskId, f: F) -> T
+where
+    F: FnOnce(&mut Labels) -> T,
+{
+    LABELS.with(|cell| {
+        let mut map = cell.borrow_mut();
+        let m = map.entry(task_id).or_default();
+        f(m)
+    })
+}
+
+/// Get a label of the given type for the specified task, if any
+#[inline]
```

Comment:
Do these inline annotations end up having any effect?

Reply:
Yeah, I wondered about that, since the compiler can figure this out on its own anyway.  Removing.